code/game/atom/atom_appearance.dm

```diff
@@ -56,6 +56,8 @@
 				continue
 			if(istext(maybe_not_an_atom) || isicon(maybe_not_an_atom))
 				continue
+			if(maybe_not_an_atom.layer < 0 && (maybe_not_an_atom.pixel_x || maybe_not_an_atom.pixel_y))
+				stack_trace("Float layer added with pixel_x and pixel_y")
```

Comment:
If we actually end up pushing this to prod (which I don't think is necessarily a good idea)
```suggestion
				stack_trace("Float layer added to [src] ([type]) with pixel_x and pixel_y set on an overlay [maybe_not_an_atom.icon]/[maybe_not_an_atom.icon_state]")
```
so parsing resulting runtimes without a debugger is easier

Reply:
Honestly I don't think pushing it to prod would be a good idea. Either should be CI compile only, or TM only